### PR TITLE
Ft/readme final

### PR DESCRIPTION
**Pull request by andersonresende**
Optional?
- Sentry configuration

Missing Sections?
- Production Deployment (Once it's finished)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Contributing
```
Comment (andersonresende) [inline CONTRIBUTING.md:1]: Need to finish this page, once I have others parts done.

File: README.md
Comment (andersonresende) [inline README.md:1]: I checked the sections in our react boilerplate and in other fastapi boilerplate.

File: README.md [hunk 2]
```diff
@@ -122,12 +151,11 @@ To run the project locally, use the following commands:
...
 #### Frontend
-1. Navigate to the `nextjs-frontend` directory.
-2. Install dependencies (if not already installed):
+1. Navigate to the `nextjs-frontend` directory and run:
    ```bash
-   npm install
+   pnpm install
    ```
-3. Start the Next.js development server:
+2. Start the Next.js development server:
```
Comment (fjsj) [inline README.md:158]: This part of running should be separate from install. Put this under the section `Running the Application`.
Also, move `Pre-Commit Setup` to the bottom after `Testing` secion.

File: README.md [hunk 3]
```diff
@@ -5,58 +5,83 @@
...
   * [If you are not using Docker](#if-you-are-not-using-docker)
     * [Backend](#backend)
     * [Frontend](#frontend)
   * [If you are using Docker](#if-you-are-using-docker)
 * [Pre-Commit Setup](#pre-commit-setup)
+  * [Installing and Activating Pre-Commit Hooks](#installing-and-activating-pre-commit-hooks)
+  * [Running Pre-Commit Checks](#running-pre-commit-checks)
+  * [Updating Pre-Commit Hooks](#updating-pre-commit-hooks)
 * [Running the Application](#running-the-application)
 * [Watchers](#watchers)
```
Comment (fjsj) [inline README.md:25]: I'm not sure if watchers is the best name for what we have. Perhaps call it hot-reload or something similar.

File: README.md [hunk 4]
```diff
@@ -5,58 +5,83 @@
...
   * [Recommended Approach](#recommended-approach-run-both-watchers-and-servers-simultaneously)
   * [Manual Execution of Watcher Commands](#manual-execution-of-watcher-commands)
+* [Testing](#testing)
+* [Alembic migrations](#alembic-migrations)
 * [Makefile](#makefile)
 * [Important Considerations](#important-considerations)
+* [Contributing](#contributing)
+* [Commercial Support](#commercial-support)
 
 ## About
```
Comment (fjsj) [inline README.md:35]: Alembic is not mentioned here.

**Review by fjsj — CHANGES_REQUESTED** [review]
Overall looking better, but I will wait for all sections to be done to give my final approval. I may edit as well after you finish all sections.

**Comment by andersonresende** [comment]
> Overall looking better, but I will wait for all sections to be done to give my final approval. I may edit as well after you finish all sections.

Thanks, I'll keep improving it.

File: README.md [hunk 5]
```diff
@@ -5,58 +5,102 @@
...
-- [Docker](https://www.docker.com/): Consistent environments for development and production  
+- [Docker](https://www.docker.com/) and [Docker Compose](https://docs.docker.com/compose/): Consistent environments for development and production  
 - [Pre-commit hooks](https://pre-commit.com/): Enforce code quality with automated checks  
 - [OpenAPI JSON schema](https://swagger.io/specification/): Centralized API documentation and client generation  
 
-With this setup, you’ll save time and maintain a seamless connection between your backend and frontend, boosting productivity and reliability.  
+With this setup, you'll save time and maintain a seamless connection between your backend and frontend, boosting productivity and reliability.
 
-## Local Setup
+## Production-Ready Authentication & Dashboard features
```
Comment (andersonresende) [inline README.md:70]: I would like to add a video or something similar to show the login/dashboard pages.

File: CONTRIBUTING.md (hunk 1 above)
Comment (fjsj) [reply CONTRIBUTING.md:1]: Track this as well please

File: README.md [hunk 6]
```diff
@@ -5,58 +5,104 @@
...
+This template comes with a pre-configured authentication system and a simple dashboard interface, allowing you to start building your application with user management features right away.
 
-### Installing Required Tools
+## Share your project!
```
Comment (fjsj) [inline README.md:75]: Move this section to after Contributing. Also, tone down the wording here, as this template/boilerplate is new:
> Several people have leveraged Vinta templates and boilerplates to start projects and startups. Starting with a solid foundation allows you to create more resilient products and focus on what really matters: discovering and delivering value to your customers. If you are one of those people, we're eager to help you even more! We can spread the word about your project across our social media platforms, giving you access to a broader audience.

File: README.md [hunk 7]
```diff
@@ -70,82 +116,135 @@ Docker is needed to run the project in a containerized environment. Follow the a
...
+Copy the `.env.example` files to `.env` and update the variables with your own values.
+```bash
+cp project_name/.env.example project_name/.env
 ```
 
 **Frontend (`nextjs-frontend/.env.local`):**
-```env
-NEXT_PUBLIC_API_BASE_URL=http://localhost:8000
-OPENAPI_OUTPUT_FILE=openapi.json
+Copy the `.env.local.example` files to `.env.local` and update the variables with your own values.
```
Comment (fjsj) [inline README.md:127]: This will hardly change. So say that.

File: README.md [hunk 8]
```diff
@@ -70,82 +116,135 @@ Docker is needed to run the project in a containerized environment. Follow the a
...
-ACCESS_SECRET_KEY=<your-secret-key>
-RESET_PASSWORD_SECRET_KEY=<your-secret-key>
-VERIFICATION_SECRET_KEY=<your-secret-key>
-OPENAPI_OUTPUT_FILE=../nextjs-frontend/openapi.json
-
-MAIL_USERNAME=<your-username>
-MAIL_PASSWORD=<your-password>
-MAIL_FROM=<your-from>
-MAIL_SERVER=<your-server>
+Copy the `.env.example` files to `.env` and update the variables with your own values.
```
Comment (fjsj) [inline README.md:121]: You have to explain what needs to be filled. Like: 
- Change DATABASE_URL if you're planning to use Docker to run the local servers
- Same for TEST_DATABASE_URL
- Generate SECRET_KEYs and save them (suggest how to generate, maybe with a python command like `python3 -c "import secrets; print(secrets.token_hex(32))"`)
- Set a SMTP mail server if you need

Comment (fjsj) [reply README.md:121]: Also, why don't we have a local test-only SMTP server?

File: README.md (hunk 7 above)
Comment (fjsj) [reply README.md:127]: I'm seeing no gain in have two env files, local and non-local.

File: README.md [hunk 9]
```diff
@@ -70,82 +116,135 @@ Docker is needed to run the project in a containerized environment. Follow the a
...
    docker compose up db
    ```
-3. Run the following command to apply database migrations:
+2. Run the following command to apply database migrations:
    ```bash
    make docker-migrate-db
    ```
-4. Start the FastAPI server:
+
+### If you are not using Docker:
```
Comment (fjsj) [inline README.md:143]: Change to: "Run the project (without Docker)"
and the other section should be: "Run the project (with Docker)"

File: README.md [hunk 10]
```diff
@@ -70,82 +116,135 @@ Docker is needed to run the project in a containerized environment. Follow the a
...
+   ```bash
    make docker-up-frontend
    ```
-   This command will automatically set up the database and other necessary containers.
 
-3. To create the database schema for the first time, run:
+- **Backend**: Access the API at `http://localhost:8000`.
+- **Frontend**: Access the web application at `http://localhost:3000`.
+
+## Watchers
```
Comment (fjsj) [inline README.md:194]: Rename this section to "Hot Reload on development". This is the common name used for this, I checked.

**Review by fjsj — CHANGES_REQUESTED** [review]
Please check comments.

File: README.md (hunk 8 above)
Comment (andersonresende) [reply README.md:115]: 1. Sure!
2. Makes sense, I'll add a local server.

Comment (andersonresende) [reply README.md:115]: Please, check if all is good now, the only things that the user needs to generate and change are the secrect keys.
I left a comment mentioning it. And there's an explanation in the .env file about the docker configuration.

File: README.md [hunk 11]
```diff
@@ -5,58 +5,100 @@
...
+  - Email-based password recovery
+- [Shadcn/ui](https://ui.shadcn.com/): Beautiful and customizable React components
 - [OpenAPI-fetch](https://github.com/Hey-AI/openapi-fetch): Fully typed client generation from OpenAPI schema  
+- [fastapi-mail](https://sabuhish.github.io/fastapi-mail/): Efficient email handling for FastAPI applications
+- [Poetry](https://python-poetry.org/): Dependency management and packaging made easy
+- [Pytest](https://docs.pytest.org/): Powerful Python testing framework
+- Code Quality Tools:
+  - [Ruff](https://github.com/astral-sh/ruff): Fast Python linter
+  - [ESLint](https://eslint.org/): JavaScript/TypeScript code quality
 - Watchers:
```
Comment (fjsj) [inline README.md:64]: Hot-reload watchers

File: README.md [hunk 12]
```diff
@@ -70,82 +112,141 @@ Docker is needed to run the project in a containerized environment. Follow the a
...
+
+### Build the project (with Docker):
+
+1. Build the backend and frontend containers:
+   ```bash
+   make docker-build-backend
+   make docker-build-frontend
+   ```
+
+## Running the Application
```
Comment (fjsj) [inline README.md:169]: I don't think we should show start-backend if the recommended way to run is in hot-reload-mode. Just show the hot-reload mode.

File: README.md [hunk 13]
```diff
@@ -70,82 +112,141 @@ Docker is needed to run the project in a containerized environment. Follow the a
...
-
-MAIL_USERNAME=<your-username>
-MAIL_PASSWORD=<your-password>
-MAIL_FROM=<your-from>
-MAIL_SERVER=<your-server>
+Copy the `.env.example` files to `.env` and update the variables with your own values.
+```bash
+cp project_name/.env.example project_name/.env
 ```
+1. You will only need to update the User Secrets. You can use the following command to generate a new secret key:
```
Comment (fjsj) [inline README.md:121]: User Secrets -> secret keys

File: README.md [hunk 14]
```diff
@@ -70,82 +112,141 @@ Docker is needed to run the project in a containerized environment. Follow the a
...
+   ```bash
+   python3 -c "import secrets; print(secrets.token_hex(32))"
+   ```
+2. The DATABASE, Email, OPENAPI, and FRONTEND_URL settings are ready to use locally.
```
Comment (fjsj) [inline README.md:125]: Email -> MAIL

File: README.md [hunk 15]
```diff
@@ -70,82 +112,141 @@ Docker is needed to run the project in a containerized environment. Follow the a
...
+3. You can check the .env.example file for more information about the variables.
 
 **Frontend (`nextjs-frontend/.env.local`):**
-```env
-NEXT_PUBLIC_API_BASE_URL=http://localhost:8000
-OPENAPI_OUTPUT_FILE=openapi.json
+Copy the `.env.example` files to `.env`. These values are unlikely to change, so you can leave them as they are.
+```bash
+cp project_name/.env.example project_name/.env
```
Comment (fjsj) [inline README.md:131]: The cp command is wrong. There's no project_name here.
I think this should be:
```
cd fastapi_backend
cp .env.example .env
```

Also fix the frontend (see below)

**Review by fjsj — CHANGES_REQUESTED** [review]
Overall LGTM, minor improvements still needed.

File: README.md (hunk 15 above)
Comment (andersonresende) [reply README.md:131]: When you use the template you have the option to set the project name. But, I can just change as you suggested.

File: README.md (hunk 12 above)
Comment (andersonresende) [reply README.md:169]: Now, the command is watch-backend and watch-frontend. What command name do you suggest?

Comment (andersonresende) [reply README.md:169]: - make backend-hotreload
- make hot-backend
- make backend-live
- make backend-run-live

Comment (fjsj) [reply README.md:169]: There's a start-backend in the README. Check it.

Comment (fjsj) [reply README.md:169]: What I mean is: we should only recommend watch mode on readme. Not  start-backend.

**Review by fjsj — APPROVED** [review]

**Pull request merged**